### PR TITLE
[DUOS-1715][risk=no] Limit info sent to users based on role

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-1715

If the user has Admin, Chair, or Member roles, show the email address of approved users. Otherwise, filter that column from the results.

### Privileged view:
Screen Shot 2022-04-28 at 3 47 07 PM

### Non-privileged view:
Screen Shot 2022-04-28 at 3 46 43 PM

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/resources/assets/paths/datasetByIdApprovedUsers.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  responses:
+    200:
+      description: Download Dataset Approved Users
+      content:
+        application/octet-stream:
+          schema:
+            type: string
+            format: binary
+    500:
+      description: Server error.
```
Comment (shaemarks) [inline src/main/resources/assets/paths/datasetByIdApprovedUsers.yaml:24]: I think there's also a 404 response option from the NotFoundException thrown when the requesting user is null (see `getDatasetApprovedUsersContent` in `DataAccessRequestService`)

File: src/main/java/org/broadinstitute/consent/http/service/DataAccessReportsParser.java [hunk 2]
```diff
@@ -52,16 +56,33 @@ public void setReviewedDARHeader(FileWriter darWriter) throws IOException {
...
+    } else {
+      builder
+          .append(HeaderDAR.NAME.getValue())
+          .append(DEFAULT_SEPARATOR)
+          .append(HeaderDAR.ORGANIZATION.getValue())
+          .append(DEFAULT_SEPARATOR)
+          .append(HeaderDAR.DAR_ID.getValue())
+          .append(DEFAULT_SEPARATOR)
+          .append(HeaderDAR.DATE_REQUEST_APPROVAL.getValue())
+          .append(END_OF_LINE);
```
Comment (shaemarks) [inline src/main/java/org/broadinstitute/consent/http/service/DataAccessReportsParser.java]: Nit: option to remove repetition between cases
```
  public String getDatasetApprovedUsersHeader(User user) {
    StringBuilder builder = new StringBuilder();
    if (user.doesUserHaveAnyRoleInSet(EnumSet.of(UserRoles.ADMIN, UserRoles.CHAIRPERSON, UserRoles.MEMBER))) {
      builder
          .append(HeaderDAR.USERNAME.getValue())
          .append(DEFAULT_SEPARATOR);
   }
      builder
          .append(HeaderDAR.NAME.getValue())
          .append(DEFAULT_SEPARATOR)
          .append(HeaderDAR.ORGANIZATION.getValue())
          .append(DEFAULT_SEPARATOR)
          .append(HeaderDAR.DAR_ID.getValue())
          .append(DEFAULT_SEPARATOR)
          .append(HeaderDAR.DATE_REQUEST_APPROVAL.getValue())
          .append(END_OF_LINE);
    
    return builder.toString();
  }
```

File: src/main/java/org/broadinstitute/consent/http/service/DataAccessReportsParser.java [hunk 3]
```diff
@@ -81,15 +102,32 @@ public void addReviewedDARLine(FileWriter darWriter, Election election, DataAcce
...
+        builder
+            .append(name)
+            .append(DEFAULT_SEPARATOR)
+            .append(institution)
+            .append(DEFAULT_SEPARATOR)
+            .append(darCode)
+            .append(DEFAULT_SEPARATOR)
+            .append(formatTimeToDate(approvalDate.getTime()))
+            .append(END_OF_LINE);
+      }
```
Comment (shaemarks) [inline src/main/java/org/broadinstitute/consent/http/service/DataAccessReportsParser.java:102]: See above comment

**Review by shaemarks — COMMENTED** [review]
Approved Users CSV correctly displays username only when current user has admin, member, or chair role(s). I left a few minor comments below.

File: src/main/java/org/broadinstitute/consent/http/service/DataAccessReportsParser.java (hunk 2 above)
Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/service/DataAccessReportsParser.java]: Nice suggestion!

**Review by shaemarks — APPROVED** [review]
Looks good! Thanks for the changes!

**Review by quazi-h — APPROVED** [review]
Changes look good to me and I was able to verify that the Username column is removed when attempting to use the dataset/{id}/approved/users without the appropriate role.

**Pull request merged**